python/tk_desktop/loading_project_widget.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+
+from sgtk.platform.qt import QtCore, QtGui
+from .ui.loading_project_widget import Ui_LoadingProjectWidget
+
+
+class LoadingProjectWidget(QtGui.QWidget):
+    """
+    Shows a Shotgun progress widget with a text box containing details of how
+    far along we are in the bootstrap process. It also supports displaying
+    an error message on top.
+    """
+
+    _SHOW_DETAILS = "show details"
```

Comment:
What do you prefer @robblau ?

<img width="427" alt="screen shot 2017-01-27 at 15 40 57" src="https://cloud.githubusercontent.com/assets/8126447/22386878/0cd90df2-e4a7-11e6-9a39-3725ae1ade68.png">